### PR TITLE
Scalability introspector + State refactor

### PR DESCRIPTION
**Pull request by tokatoka**

### CODE REVIEW

File: libafl_libfuzzer/libafl_libfuzzer_runtime/src/feedbacks.rs [hunk 1]
```diff
@@ -113,7 +113,7 @@ impl LibfuzzerCrashCauseFeedback {
 
 impl<S> Feedback<S> for LibfuzzerCrashCauseFeedback
 where
-    S: UsesInput<Input = BytesInput> + HasClientPerfMonitor,
```
Comment (tokatoka) [inline libafl_libfuzzer/libafl_libfuzzer_runtime/src/feedbacks.rs:116]: I don't understand why this code compiles...?
@addisoncrump 

Essentially we changed UsesInput to State.
But in UsesInput you can force(?) its `Input` to be `BytesInput`.
Now we're using `State` but it doesn't know anything about `Input` type.
Why this works????

File: libafl/src/feedbacks/nautilus.rs [hunk 2]
```diff
@@ -82,10 +82,7 @@ impl<'a, S> Named for NautilusFeedback<'a, S> {
 
 impl<'a, S> Feedback<S> for NautilusFeedback<'a, S>
 where
-    S: HasMetadata
```
Comment (tokatoka) [inline libafl/src/feedbacks/nautilus.rs:85]: Same pattern is in here..?

Would this be a problem?

File: libafl_libfuzzer/libafl_libfuzzer_runtime/src/feedbacks.rs (hunk 1 above)
Comment (addisoncrump) [reply libafl_libfuzzer/libafl_libfuzzer_runtime/src/feedbacks.rs:116]: State<Input = BytesInput>

File: libafl/src/feedbacks/nautilus.rs (hunk 2 above)
Comment (addisoncrump) [reply libafl/src/feedbacks/nautilus.rs:85]: This one is fine -- anywhere you see `<Input = ...>` automatically gets flattened out to UsesInput's `Input`.

File: libafl_libfuzzer/libafl_libfuzzer_runtime/src/feedbacks.rs (hunk 1 above)
Comment (tokatoka) [reply libafl_libfuzzer/libafl_libfuzzer_runtime/src/feedbacks.rs:116]: ok. it just compiles..
but why syntax like 
```
State<Input = BytesInput>
```

Shouldn't this `Input` a associate type of `State`?
but the trait `State` is this:
```
pub trait State:
    UsesInput + Serialize + DeserializeOwned + MaybeHasClientPerfMonitor + MaybeHasScalabilityMonitor
{
}
```
And there's no such thing like `Input` in this `State` definition

**Pull request merged**